`admission-webhook` rock lacks executable on path `/webhook`

### Bug Description

It seems that the `admission-webhook` rock doesn't have the same file structure as the upstream image. Specifically, it is missing the executable on the path "/webhook", which is used as an entry point for `admission-webhook-operator`, as seen in [this line](https://github.com/canonical/admission-webhook-operator/blob/main/src/charm.py#L53).

This issue when encountered while working on [this issue](https://github.com/canonical/admission-webhook-operator/issues/151).

### To Reproduce

- Run bash inside the rock container with:
```
docker run -it --entrypoint sh charmedkubeflow/admission-webhook:v1.9.0-9c8b687
```
Then run 
```
find . -name "webhook"
```
No results are returned, whereas the upstream image should contain an executable on the path `/webhook`

### Environment

Full image name: `charmedkubeflow/admission-webhook:v1.9.0-9c8b687`

### Relevant Log Output

```shell
tox -e integration -- --model kubeflow --keep-models
integration: commands[0]> pytest -v --tb native --asyncio-mode=auto /home/ubuntu/admission-webhook-operator/tests/integration --log-cli-level=INFO -s --model kubeflow --keep-models
============================= test session starts =============================
platform linux -- Python 3.8.19, pytest-8.2.2, pluggy-1.5.0 -- /home/ubuntu/admission-webhook-operator/.tox/integration/bin/python
cachedir: .tox/integration/.pytest_cache
rootdir: /home/ubuntu/admission-webhook-operator
configfile: pyproject.toml
plugins: operator-0.35.0, asyncio-0.21.2, anyio-4.4.0
asyncio: mode=auto
collected 4 items                                                             

tests/integration/test_charm.py::test_build_and_deploy 
------------------------------- live log setup --------------------------------
INFO     pytest_operator.plugin:plugin.py:734 Connecting to existing model uk8sx:kubeflow on unspecified cloud
-------------------------------- live log call --------------------------------
INFO     pytest_operator.plugin:plugin.py:575 Using tmp_path: /home/ubuntu/admission-webhook-operator/.tox/integration/tmp/pytest/kubeflow0
INFO     pytest_operator.plugin:plugin.py:1083 Building charm admission-webhook
INFO     pytest_operator.plugin:plugin.py:1088 Built charm admission-webhook in 1354.16s
INFO     test_charm:test_charm.py:33 Built charm /home/ubuntu/admission-webhook-operator/.tox/integration/tmp/pytest/kubeflow0/charms/admission-webhook_ubuntu-20.04-amd64.charm
INFO     juju.model:model.py:2097 Deploying local:admission-webhook-0
INFO     charmed_kubeflow_chisme.testing.cos_integration:cos_integration.py:65 deploying grafana-agent-k8s from latest/stable channel
INFO     juju.model:model.py:2097 Deploying ch:amd64/jammy/grafana-agent-k8s-45
INFO     charmed_kubeflow_chisme.testing.cos_integration:cos_integration.py:95 Adding relation: admission-webhook:logging and grafana-agent-k8s:logging-provider
INFO     juju.model:model.py:2971 Waiting for model:
  grafana-agent-k8s/0 [allocating] waiting: installing agent
INFO     juju.model:model.py:2971 Waiting for model:
  grafana-agent-k8s/0 [executing] maintenance: installing charm software
PASSED
tests/integration/test_charm.py::test_logging 
-------------------------------- live log call --------------------------------
INFO     charmed_kubeflow_chisme.testing.cos_integration:cos_integration.py:138 found relations [<Relation id=1 admission-webhook:logging grafana-agent-k8s:logging-provider>] for grafana-agent-k8s:logging
INFO     charmed_kubeflow_chisme.testing.cos_integration:cos_integration.py:238 running cmd `relation-get --format=yaml -r 1 - grafana-agent-k8s/0` on unit grafana-agent-k8s/0
PASSED
tests/integration/test_charm.py::test_is_active FAILED
tests/integration/test_charm.py::test_namespace_selector_poddefault_service_account_token_mounted[./tests/integration/poddefault_test_workloads.yaml] 
------------------------------- live log setup --------------------------------
INFO     httpx:_client.py:1026 HTTP Request: PATCH https://127.0.0.1:16443/api/v1/namespaces/test-admission-webhook-user-namespace?fieldManager=test "HTTP/1.1 200 OK"
INFO     httpx:_client.py:1026 HTTP Request: PATCH https://127.0.0.1:16443/apis/kubeflow.org/v1alpha1/namespaces/test-admission-webhook-user-namespace/poddefaults/access-ml-pipeline?fieldManager=test "HTTP/1.1 201 Created"
INFO     httpx:_client.py:1026 HTTP Request: PATCH https://127.0.0.1:16443/api/v1/namespaces/test-admission-webhook-user-namespace/pods/testpod?fieldManager=test "HTTP/1.1 500 Internal Server Error"
ERROR
------------------------------ live log teardown ------------------------------
INFO     pytest_operator.plugin:plugin.py:862 Model status:

Model     Controller  Cloud/Region      Version  SLA          Timestamp
kubeflow  uk8sx       my-k8s/localhost  3.4.5    unsupported  18:16:35+03:00

App                Version  Status   Scale  Charm              Channel        Rev  Address        Exposed  Message
admission-webhook           waiting      1  admission-webhook                   0  10.152.183.78  no       installing agent
grafana-agent-k8s  0.32.1   waiting      1  grafana-agent-k8s  latest/stable   45  10.152.183.58  no       installing agent

Unit                  Workload  Agent  Address      Ports  Message
admission-webhook/0*  blocked   idle   10.1.90.177         Failed to replan
grafana-agent-k8s/0*  blocked   idle   10.1.90.179         grafana-cloud-config: off, logging-consumer: off

INFO     pytest_operator.plugin:plugin.py:868 Juju error logs:

unit-admission-webhook-0: 18:15:18 ERROR unit.admission-webhook/0.juju-log Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/charmed_kubeflow_chisme/pebble/_update_layer.py", line 31, in update_layer
    container.replan()
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/ops/model.py", line 2252, in replan
    self._pebble.replan_services()
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/ops/pebble.py", line 1993, in replan_services
    return self._services_action('replan', [], timeout, delay)
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/ops/pebble.py", line 2090, in _services_action
    raise ChangeError(change.err, change)
ops.pebble.ChangeError: cannot perform the following tasks:
- Start service "admission-webhook" (cannot start service: fork/exec /webhook: no such file or directory)
----- Logs from task 0 -----
2024-08-30T15:15:18Z ERROR cannot start service: fork/exec /webhook: no such file or directory
-----

unit-admission-webhook-0: 18:15:18 ERROR unit.admission-webhook/0.juju-log Failed to handle <InstallEvent via AdmissionWebhookCharm/on/install[1]> with error: Failed to replan
unit-admission-webhook-0: 18:15:20 ERROR unit.admission-webhook/0.juju-log Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/charmed_kubeflow_chisme/pebble/_update_layer.py", line 31, in update_layer
    container.replan()
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/ops/model.py", line 2252, in replan
    self._pebble.replan_services()
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/ops/pebble.py", line 1993, in replan_services
    return self._services_action('replan', [], timeout, delay)
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/ops/pebble.py", line 2090, in _services_action
    raise ChangeError(change.err, change)
ops.pebble.ChangeError: cannot perform the following tasks:
- Start service "admission-webhook" (cannot start service: fork/exec /webhook: no such file or directory)
----- Logs from task 0 -----
2024-08-30T15:15:20Z ERROR cannot start service: fork/exec /webhook: no such file or directory
-----

unit-admission-webhook-0: 18:15:20 ERROR unit.admission-webhook/0.juju-log Failed to handle <PebbleReadyEvent via AdmissionWebhookCharm/on/admission_webhook_pebble_ready[16]> with error: Failed to replan
unit-admission-webhook-0: 18:15:22 ERROR unit.admission-webhook/0.juju-log Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/charmed_kubeflow_chisme/pebble/_update_layer.py", line 31, in update_layer
    container.replan()
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/ops/model.py", line 2252, in replan
    self._pebble.replan_services()
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/ops/pebble.py", line 1993, in replan_services
    return self._services_action('replan', [], timeout, delay)
  File "/var/lib/juju/agents/unit-admission-webhook-0/charm/venv/ops/pebble.py", line 2090, in _services_action
    raise ChangeError(change.err, change)
ops.pebble.ChangeError: cannot perform the following tasks:
- Start service "admission-webhook" (cannot start service: fork/exec /webhook: no such file or directory)
----- Logs from task 0 -----
2024-08-30T15:15:22Z ERROR cannot start service: fork/exec /webhook: no such file or directory
-----

unit-admission-webhook-0: 18:15:22 ERROR unit.admission-webhook/0.juju-log Failed to handle <ConfigChangedEvent via AdmissionWebhookCharm/on/config_changed[21]> with error: Failed to replan

INFO     pytest_operator.plugin:plugin.py:947 Forgetting model main...


=================================== ERRORS ====================================
_ ERROR at setup of test_namespace_selector_poddefault_service_account_token_mounted[./tests/integration/poddefault_test_workloads.yaml] _
Traceback (most recent call last):
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/lightkube/core/generic_client.py", line 188, in raise_for_status
    resp.raise_for_status()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/httpx/_models.py", line 761, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Server error '500 Internal Server Error' for url 'https://127.0.0.1:16443/api/v1/namespaces/test-admission-webhook-user-namespace/pods/testpod?fieldManager=test'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 241, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 139, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/unraisableexception.py", line 85, in pytest_runtest_setup
    yield from unraisable_exception_runtest_hook()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/unraisableexception.py", line 65, in unraisable_exception_runtest_hook
    yield
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/logging.py", line 844, in pytest_runtest_setup
    yield from self._runtest_for(item, "setup")
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/logging.py", line 833, in _runtest_for
    yield
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/capture.py", line 873, in pytest_runtest_setup
    return (yield)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/threadexception.py", line 82, in pytest_runtest_setup
    yield from thread_exception_runtest_hook()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/threadexception.py", line 63, in thread_exception_runtest_hook
    yield
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 159, in pytest_runtest_setup
    item.session._setupstate.setup(item)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 517, in setup
    raise exc
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 514, in setup
    col.setup()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/python.py", line 1635, in setup
    self._request._fillfixtures()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/fixtures.py", line 686, in _fillfixtures
    item.funcargs[argname] = self.getfixturevalue(argname)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/fixtures.py", line 521, in getfixturevalue
    fixturedef = self._get_active_fixturedef(argname)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/fixtures.py", line 606, in _get_active_fixturedef
    fixturedef.execute(request=subrequest)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/fixtures.py", line 1076, in execute
    result = ihook.pytest_fixture_setup(fixturedef=self, request=request)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 182, in _multicall
    return outcome.get_result()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_result.py", line 100, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 167, in _multicall
    teardown.throw(outcome._exception)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/setuponly.py", line 36, in pytest_fixture_setup
    return (yield)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/fixtures.py", line 1125, in pytest_fixture_setup
    result = call_fixture_func(fixturefunc, request, kwargs)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/fixtures.py", line 880, in call_fixture_func
    fixture_result = next(generator)
  File "/home/ubuntu/admission-webhook-operator/tests/integration/test_charm.py", line 94, in kubernetes_workloads
    apply_many(lightkube_client, workloads, "test")
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/charmed_kubeflow_chisme/lightkube/batch/_many.py", line 72, in apply_many
    returns[i] = client.apply(
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/lightkube/core/client.py", line 457, in apply
    return self.patch(type(obj), name, obj, namespace=namespace,
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/lightkube/core/client.py", line 325, in patch
    return self._client.request("patch", res=res, name=name, namespace=namespace, obj=obj,
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/lightkube/core/generic_client.py", line 245, in request
    return self.handle_response(method, resp, br)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/lightkube/core/generic_client.py", line 196, in handle_response
    self.raise_for_status(resp)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/lightkube/core/generic_client.py", line 190, in raise_for_status
    raise transform_exception(e)
lightkube.core.exceptions.ApiError: Internal error occurred: failed calling webhook "admission-webhook.kubeflow.org": failed to call webhook: Post "https://admission-webhook.kubeflow.svc:4443/apply-poddefault?timeout=10s": dial tcp 10.152.183.78:4443: connect: connection refused
----------------------------- Captured log setup ------------------------------
INFO     httpx:_client.py:1026 HTTP Request: PATCH https://127.0.0.1:16443/api/v1/namespaces/test-admission-webhook-user-namespace?fieldManager=test "HTTP/1.1 200 OK"
INFO     httpx:_client.py:1026 HTTP Request: PATCH https://127.0.0.1:16443/apis/kubeflow.org/v1alpha1/namespaces/test-admission-webhook-user-namespace/poddefaults/access-ml-pipeline?fieldManager=test "HTTP/1.1 201 Created"
INFO     httpx:_client.py:1026 HTTP Request: PATCH https://127.0.0.1:16443/api/v1/namespaces/test-admission-webhook-user-namespace/pods/testpod?fieldManager=test "HTTP/1.1 500 Internal Server Error"
================================== FAILURES ===================================
_______________________________ test_is_active ________________________________
Traceback (most recent call last):
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 241, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 139, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/threadexception.py", line 87, in pytest_runtest_call
    yield from thread_exception_runtest_hook()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/threadexception.py", line 63, in thread_exception_runtest_hook
    yield
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/unraisableexception.py", line 90, in pytest_runtest_call
    yield from unraisable_exception_runtest_hook()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/unraisableexception.py", line 65, in unraisable_exception_runtest_hook
    yield
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/logging.py", line 850, in pytest_runtest_call
    yield from self._runtest_for(item, "call")
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/logging.py", line 833, in _runtest_for
    yield
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/capture.py", line 878, in pytest_runtest_call
    return (yield)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/skipping.py", line 257, in pytest_runtest_call
    return (yield)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 183, in pytest_runtest_call
    raise e
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 173, in pytest_runtest_call
    item.runtest()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/python.py", line 1632, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 182, in _multicall
    return outcome.get_result()
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/
site-packages/pluggy/_result.py", line 100, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/_pytest/python.py", line 162, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/pytest_asyncio/plugin.py", line 529, in inner
    _loop.run_until_complete(task)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/ubuntu/admission-webhook-operator/tests/integration/test_charm.py", line 57, in test_is_active
    await ops_test.model.wait_for_idle(
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 2964, in wait_for_idle
    _raise_for_status(blocks, "blocked")
  File "/home/ubuntu/admission-webhook-operator/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 2869, in _raise_for_status
    raise error_type("{}{} in {}: {}".format(
juju.errors.JujuUnitError: Unit in blocked: admission-webhook/0
=========================== short test summary info ===========================
FAILED tests/integration/test_charm.py::test_is_active - juju.errors.JujuUnitError: Unit in blocked: admission-webhook/0
ERROR tests/integration/test_charm.py::test_namespace_selector_poddefault_service_account_token_mounted[./tests/integration/poddefault_test_workloads.yaml] - lightkube.core.exceptions.ApiError: Internal error occurred: failed callin...
============== 1 failed, 2 passed, 1 error in 1445.77s (0:24:05) ==============
integration: exit 1 (1446.47 seconds) /home/ubuntu/admission-webhook-operator> pytest -v --tb native --asyncio-mode=auto /home/ubuntu/admission-webhook-operator/tests/integration --log-cli-level=INFO -s --model kubeflow --keep-models pid=2185804
  integration: FAIL code 1 (1446.50=setup[0.04]+cmd[1446.47] seconds)
  evaluation failed :( (1446.57 seconds)
```


### Additional Context

_No response_

--- a/admission-webhook/rockcraft.yaml
+++ b/admission-webhook/rockcraft.yaml
@@ -1,3 +1,4 @@
+# From https://github.com/kubeflow/kubeflow/blob/v1.9-branch/components/admission-webhook/Dockerfile
 name: admission-webhook
 base: ubuntu@20.04
 version: v1.9.0
@@ -16,7 +17,7 @@ platforms:
 services:
   base-admission-webhook:
     override: merge
-    command: "admission-webhook"
+    command: "/webhook"
     startup: enabled
     user: ubuntu
 
@@ -27,14 +28,23 @@ parts:
       mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
       (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
 
+  # The go plugin uses go install to build and install the "admission-webhook" binary at
+  # /bin/admission-webhook, which is undesirable as it is expected to be in /webhook.
+  # The go plugin does not offer any mechanism for passing custom install paths, so we have
+  # to use override-build instead to make sure the binary is placed in the expected path.
   admission-webhook:
     plugin: go
     build-snaps:
       - go/1.21/stable
     source: https://github.com/kubeflow/kubeflow.git
-    source-tag: v1.9.0
-    source-depth: 1
-    source-subdir: components/admission-webhook
+    source-tag: v1.9-branch
+    build-environment:
+      - CGO_ENABLED: 0
+      - GOOS: linux
+    override-build: |
+      cd components/admission-webhook
+      go build -o webhook -a .
+      cp webhook $CRAFT_PART_INSTALL/webhook
 
   non-root-user:
     plugin: nil

--- a/admission-webhook/rockcraft.yaml
+++ b/admission-webhook/rockcraft.yaml
@@ -43,8 +43,7 @@ parts:
       - GOOS: linux
     override-build: |
       cd components/admission-webhook
-      go build -o webhook -a .
-      cp webhook $CRAFT_PART_INSTALL/webhook
+      go build -o $CRAFT_PART_INSTALL/webhook -a .
 
   non-root-user:
     plugin: nil

--- a/admission-webhook/tests/test_rock.py
+++ b/admission-webhook/tests/test_rock.py
@@ -49,7 +49,7 @@ def test_rock(rock_test_env):
             "exec",
             "ls",
             "-la",
-            "/bin/admission-webhook",
+            "/webhook",
         ],
         check=True,
     )

